core/iwasm/common/wasm_memory.c

```diff
@@ -167,3 +167,16 @@ wasm_runtime_free(void *ptr)
 {
     wasm_runtime_free_internal(ptr);
 }
+
+extern void *
+gc_heap_stats(void *heap, uint32 *stats, int size);
```

Comment:
Not a good coding custom, had better implement the API in mem alloc layer, how about:
define struct and implement API in core/shared/mem-alloc/mem_alloc.{hc}:
```
typedef struct mem_alloc_info_t {
        uint32_t total_size;
        uint32_t total_free_size;
        uint32_t highmark_size;
} mem_alloc_info_t;

bool
mem_allocator_get_alloc_info(mem_allocator_t allocator, mem_alloc_info_t *mem_alloc_info);
```

Reply:
This typedef is dupped with wasm_export.h

Reply:
How about using a macro to control it (like MEM_ALLOC_OPTION_DEFINED): in wasm_export.h and mem_alloc.h, add:
```C
#ifndef MEM_ALLOC_INFO_DEFINED
#define MEM_ALLOC_INFO_DEFINED
typedef struct mem_alloc_info_t {
        uint32_t total_size;
        uint32_t total_free_size;
        uint32_t highmark_size;
} mem_alloc_info_t;
#endif
```